tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/ConfidentialClientApplicationBuilderTests.cs

```diff
@@ -154,6 +155,42 @@ public void TestConstructor_WithRedirectUri()
             Assert.AreEqual(RedirectUri, cca.AppConfig.RedirectUri);
         }
 
+        [TestMethod]
+        public void TestConstructor_WithNullRedirectUri()
+        {
+            var cca = ConfidentialClientApplicationBuilder
+                      .Create(MsalTestConstants.ClientId).WithRedirectUri(null).Build();
+
+            Assert.AreEqual(Constants.DefaultRedirectUri, cca.AppConfig.RedirectUri);
```

Comment:
I don't think that this would be good that if developers forget to set the RedirectUri in Authorization code flow or OBO, the Redirect Uri be the one representing public client apps. Given that the STS determines the type of app based on the Redirect URI ….
For confidential client apps, I'd prefer we have another constant for instance `https://replyUrlNotSet` ?